src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -334,15 +369,26 @@ private static ImmutableArray<ValidationError> Validate(params BenchmarkRunInfo[
             {
                 var joinedCases = benchmarks.SelectMany(b => b.BenchmarksCases).ToArray();
 
-                validationErrors.AddRange(
-                    ConfigCompatibilityValidator
-                        .FailOnError
-                        .Validate(new ValidationParameters(joinedCases, null))
-                    );
+                var compatibilityValidationErrors = ConfigCompatibilityValidator
+                    .FailOnError
+                    .Validate(new ValidationParameters(joinedCases, null));
+
+                foreach (var error in compatibilityValidationErrors)
+                {
+                    validationErrors.Add(error);
```

Comment:
you have achieved exactly the same effect, but with more code. Did you intend to raise some events here?